### PR TITLE
Say the valid IP range in IP errors

### PR DESCRIPTION
**Pull request by thockin**
This is now the same as portallocator.

@ericchiang @xiangpengzhao 

xref #50274

### CODE REVIEW

**Comment by apelisse** [comment]
is this supposed to fix #50274? Do we want it in 1.8?

**Comment by thockin** [comment]
Just clarifying the error when it happens

File: pkg/registry/core/service/ipallocator/allocator.go [hunk 1]
```diff
@@ -40,11 +40,18 @@ type Interface interface {
...
 	ErrAllocated         = errors.New("provided IP is already allocated")
 	ErrMismatchedNetwork = errors.New("the provided network does not match the current range")
 )
 
+type ErrNotInRange struct {
+	ValidRange string
+}
+
+func (e *ErrNotInRange) Error() string {
+	return fmt.Sprintf("provided IP is not in the valid range. The range of valid IPs is %s", e.ValidRange)
```
Comment (apelisse) [inline pkg/registry/core/service/ipallocator/allocator.go]: Should this print the provided IP?

Comment (xiangpengzhao) [reply pkg/registry/core/service/ipallocator/allocator.go]: No, the provided IP is printed outside this error. See the current message:

```
Message: "Invalid value: \"127.1.2.3\": provided IP is not in the valid range",
```

The PR is expected to provide the valid IP range when we get an error of `provided IP is not in the valid range` for debugging.

Comment (apelisse) [reply pkg/registry/core/service/ipallocator/allocator.go]: Thanks for clarifying, feel free to lgtm. We need to move forward with the reference issue quickly!

**Comment by xiangpengzhao** [comment]
LGTM. will leave to other assignees for further review.

**Comment by apelisse** [comment]
Thanks, let's get it merged now rather than later.

/lgtm

**Comment by xiangpengzhao** [comment]
ref: #52759

**Pull request merged**